### DISCUSSION TITLE
Problem running itch.io version on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by pwiecz** [cxong/cdogs-sdl#844] [General]
I wanted badly to play the good old Cyberdogs on CDogs engine, but running the itch version did not work for me.
I'm not sure if it's the correct place to report an issue with the itch.io version, but it seems to have been built with a perhaps a newer version of libsdl2_mixer than the one provide in the current LTS release of Ubuntu (which is 2.0.4)
When I try running it it complains with error:
`./cdogs-sdl: undefined symbol: Mix_PlayChannel`
I've managed to build it from sources on my machine, though.
Btw, I've had to  create custom FindSDL2_image.cmake and FindSDL2_mixer.cmake files, as they did not seem to be provided with SDL2.

Thanks for all the fun!

**Comment by cxong** [comment]
Thanks for the info, I think the linux binary is not the best way to play on linux due to all the different distros and packaging systems. C-Dogs SDL is available on debian and a few other places so installing it from a package manager would be the way to go instead of itch.

FYI the build system uses homebrew to get the latest SDL libraries - the ones available in LTS releases are often old and sometimes we want to take advantage of new SDL features: https://github.com/cxong/cdogs-sdl/blob/master/.github/workflows/cmake.yml#L66

**Comment by milleniumbug** [comment]
The issue is not with whether you "want" to take advantage of new SDL features, but whether you actually use them. Given that it actually **compiles and runs** (aside the whole "can't find SDL2 modules" thing, which seems to be solve'able through https://github.com/aminosbh/sdl2-cmake-modules) on Linux Mint 21.3 and Ubuntu 22 implies you don't seem to be using them right now. 

Regarding `Mix_PlayChannel` specifically - the reason it's missing is because in SDL2_mixer 2.6.0 and later it's a function, and before it was a macro that calls `Mix_PlayChannelTimed(channel, chunk, loops, -1)`. Since no ABI change was made to `Mix_PlayChannelTimed`, it means the binaries compiled against older SDL2_mixer will run just fine on the latest SDL2_mixer.

The changes I needed to do to make it run on Linux Mint 21.3 are here https://github.com/milleniumbug/cdogs-sdl/commit/c47ed93030cf75db887d8c7bc5b6e7b2cd41f21b
